R/zchunk_LA1011.en_bal_adj.R

```diff
@@ -1,88 +1,214 @@
+
+    EIA_TOT_intlship_kbbld %>%
+      gather(year, value, -Country) %>%
+      mutate(year = sub("X", "", year)) %>%
+      mutate(year = as.integer(year)) -> EIA_TOT_intlship_kbbld
+
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+
+    # MODIFICATIONS TO IEA ENERGY BALANCES
+    # Subset only the relevant years and combine OECD with non-OECD
+    # Replacing IEA estimates of international shipping fuel consumption with EIA estimates
+    # First, convert available data to EJ per year of total refined liquid products
+    EIA_RFO_intlship_kbbld %>%
+      filter(year %in% HISTORICAL_YEARS) %>%
+      select(year) %>%
+      distinct(year) -> EIA_years
+
+    EIA_RFO_intlship_kbbld %>%
+      inner_join(EIA_years, by = "year") %>% #inner joining to only keep years from EIA_years
+      mutate(value = value * CONV_KBBL_BBL * CONV_BBL_TONNE_RFO *
+               CONV_TONNE_GJ_RFO * CONV_GJ_EJ / CONV_DAYS_YEAR ) -> L1011.in_EJ_ctry_intlship_RFO_Yh
+
+    EIA_RFO_intlship_kbbld %>%
+      left_join(EIA_TOT_intlship_kbbld %>%
+                  rename(value_TOT = value), by = c("year", "Country")) %>%
+      inner_join(EIA_years, by = "year") %>%
+      mutate(value_diff = value_TOT - value) %>%
+      select(Country, year, value_diff) %>%
+      mutate(value = value_diff * CONV_KBBL_BBL * CONV_BBL_TONNE_DISTILLATE *
+               CONV_TONNE_GJ_RFO * CONV_GJ_EJ / CONV_DAYS_YEAR) %>%
+      select(Country, year, value) -> L1011.in_EJ_ctry_intlship_distillate_Yh
+
+    L1011.in_EJ_ctry_intlship_distillate_Yh %>%
+      left_join(L1011.in_EJ_ctry_intlship_RFO_Yh %>%
+                  rename(value_RFO = value), by = c("year", "Country")) %>%
+      mutate(value = value + value_RFO) %>%
+      select(Country, year, value) -> L1011.in_EJ_ctry_intlship_TOT_Yh
+
+    # Several countries blink in and out of the time series. Only changing Russia because it's the only one that is a really large amount of fuel
+    L1011.in_EJ_ctry_intlship_TOT_Yh %>%
+      mutate(value = if_else(is.na(value), 0, value)) -> L1011.in_EJ_ctry_intlship_TOT_Yh
+
+    L1011.in_EJ_ctry_intlship_TOT_Yh %>%
+      filter(Country == "Russia" & year %in% 2005 ) %>%
```

Comment:
Here and below: lots of hardcoded years. Why? What's special about these numbers?